src/main/java/org/folio/rest/impl/EcsTlrSettingsService.java

```diff
@@ -0,0 +1,76 @@
+package org.folio.rest.impl;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.rest.impl.UrlPath.CIRCULATION_SETTINGS_STORAGE_URL_PATH;
+import static org.folio.rest.impl.UrlPath.ECS_TLR_SETTINGS_URL_PATH;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.integration.http.ResponseInterpreter;
+import org.folio.integration.http.VertxOkapiHttpClient;
+import org.folio.patron.rest.exceptions.HttpException;
+
+import io.vertx.core.json.JsonObject;
+
+public class EcsTlrSettingsService {
+
+  private static final Logger logger = LogManager.getLogger();
+  private static final String ECS_TLR_FEATURE_KEY = "ecsTlrFeatureEnabled";
+  private static final String CIRCULATION_SETTINGS_KEY = "circulationSettings";
+  private static final int FIRST_POSITION_INDEX = 0;
+  private static final String VALUE_KEY = "value";
+  private static final String ENABLED_KEY = "enabled";
+
+  public CompletableFuture<Boolean> isEcsTlrFeatureEnabled(VertxOkapiHttpClient httpClient,
+    Map<String, String> okapiHeaders) {
+
+    logger.debug("isEcsTlrFeatureEnabled:: trying to get isEcsTlrFeatureEnabled from mod-tlr");
+    return httpClient.get(ECS_TLR_SETTINGS_URL_PATH, okapiHeaders)
+      .thenApply(ResponseInterpreter::verifyAndExtractBody)
+      .exceptionally(this::handleErrorFromModTlr)
+      .thenCompose(body -> getEcsTlrFeatureValue(body, httpClient, okapiHeaders));
+  }
+
+  private JsonObject handleErrorFromModTlr(Throwable throwable) {
+    if (throwable instanceof HttpException) {
+      logger.warn("handleErrorFromModTlr:: failed to fetch ECS TLR settings from mod-tlr",
+        throwable);
+      return null;
+    }
+    logger.error(throwable);
+    throw new RuntimeException(throwable);
+  }
+
+  private CompletableFuture<Boolean> getEcsTlrFeatureValue(JsonObject body,
+    VertxOkapiHttpClient client, Map<String, String> okapiHeaders) {
+
+    return Objects.nonNull(body) && body.containsKey(ECS_TLR_FEATURE_KEY) ?
+      completedFuture(body.getBoolean(ECS_TLR_FEATURE_KEY)) :
+      getCirculationStorageEcsTlrFeatureValue(client, okapiHeaders);
+  }
+
+  private CompletableFuture<Boolean> getCirculationStorageEcsTlrFeatureValue(
+    VertxOkapiHttpClient client, Map<String, String> okapiHeaders) {
+
+    logger.debug("getCirculationStorageEcsTlrFeatureValue:: trying to get isEcsTlrFeatureEnabled");
+    return client.get(CIRCULATION_SETTINGS_STORAGE_URL_PATH, okapiHeaders)
+      .thenApply(ResponseInterpreter::verifyAndExtractBody)
+      .thenApply(this::getCirculationStorageEcsTlrFeatureValue);
+  }
+
+  private Boolean getCirculationStorageEcsTlrFeatureValue(JsonObject body) {
```

Comment:
Should return `boolean` instead of `Boolean`.

Reply:
replaced